tests/PHPStan/Rules/Comparison/ImpossibleCheckTypeFunctionCallRuleTest.php

```diff
@@ -1102,4 +1102,19 @@ public function testAlwaysTruePregMatch(): void
 		$this->analyse([__DIR__ . '/data/always-true-preg-match.php'], []);
 	}
 
+	public function testBug11799(): void
+	{
+		$this->checkAlwaysTrueCheckTypeFunctionCall = true;
+		$this->treatPhpDocTypesAsCertain = true;
+		$this->analyse([__DIR__ . '/data/bug-11799.php'], [
+			[
+				"Call to function in_array() with arguments string, array{'publishDate', 'approvedAt', 'allowedValues'} and true will always evaluate to false.",
+				11,
+				"• 'publishDate' is not lowercase.
+• 'approvedAt' is not lowercase.
+• 'allowedValues' is not lowercase.",
```

Comment:
Since the comparison is between `'publishDate'|'approvedAt'|'allowedValues'` and `lowercase string`, and because of how `UnionType::isSuperTypeOfWithReason` works
```
$result = IsSuperTypeOfResult::createNo()->or(...array_map(static fn (Type $innerType) => $innerType->isSuperTypeOfWithReason($otherType), $this->types));
		if ($result->yes()) {
			return $result;
		}
```
we have a list with three reasons.

Is it ok for you @ondrejmirtes ?